### DISCUSSION TITLE
how should i loading online image?

### DISCUSSION THREAD

**Discussion by michael-eddy** [AvaloniaUI/Avalonia#8164] [Q&A]
now i just using httpclient download the image stream.
than create a bitmap object and set the source value for `Image`,
but if the image is large(include size or memory) the ram is not enough to use.
there's my program within manga client,
it's not a good experience small ram device.
it's that any Suggestion?

**Comment by timunie** [comment]
Hi @michael-eddy 

1. Use async loading of images https://docs.avaloniaui.net/docs/data-binding/binding-to-tasks-and-observables#example-2-binding-to-a-task
2. Reduce the resulting size via `Bitmap.DecodeToWidth`, see: https://docs.avaloniaui.net/tutorials/music-store-app/displaying-images#displaying-album-cover-images


Happy coding
Tim

**Reply by michael-eddy** [reply]
> Hi @michael-eddy
> 
> 1. Use async loading of images https://docs.avaloniaui.net/docs/data-binding/binding-to-tasks-and-observables#example-2-binding-to-a-task
> 2. Reduce the resulting size via `Bitmap.DecodeToWidth`, see: https://docs.avaloniaui.net/tutorials/music-store-app/displaying-images#displaying-album-cover-images
> 
> Happy coding Tim

thanks,but which way that binding task from code?
